File: android/src/main/java/com/github/quadflask/react/navermap/RNNaverMapViewProps.java
package com.github.quadflask.react.navermap;

import android.view.View;

import com.naver.maps.geometry.LatLng;
import com.naver.maps.geometry.LatLngBounds;
import com.naver.maps.map.NaverMap;

public interface RNNaverMapViewProps {
    String[] EVENT_NAMES = new String[]{
            "onInitialized",
            "onCameraChange",
            "onMapClick",
            "onTouch"
    };

    void onStop();

    void setCenter(LatLng latLng, Double zoom);

    void setCenter(LatLng latLng, Double zoom, Double tilt, Double bearing);

    void zoomTo(LatLngBounds latLngBounds, int paddingInPx);

    void setTilt(int tilt);

    void setBearing(int bearing);

    void setZoom(float zoom);

    void setMapPadding(int left, int top, int right, int bottom);

    void onInitialized();

    void showsMyLocationButton(boolean show);

    void setCompassEnabled(boolean show);

    void setScaleBarEnabled(boolean show);

    void setZoomControlEnabled(boolean show);

    void setLocationTrackingMode(int mode);

    void setMapType(NaverMap.MapType value);

    void setBuildingHeight(float height);

    void setMinZoom(float minZoomLevel);

    void setMaxZoom(float maxZoomLevel);

    void setLayerGroupEnabled(String layerGroup, boolean enable);

    void setNightModeEnabled(boolean enable);

    void setLogoMargin(int left, int top, int right, int bottom);

    void setLogoGravity(int gravity);

    void setScrollGesturesEnabled(boolean enabled);

    void setZoomGesturesEnabled(boolean enabled);

    void setTiltGesturesEnabled(boolean enabled);

    void setRotateGesturesEnabled(boolean enabled);

    void setStopGesturesEnabled(boolean enabled);

    void setLiteModeEnabled(boolean enabled);

    void moveCameraFitBound(LatLngBounds bounds, int left, int top, int right, int bottom);

    void addFeature(View child, int index);

    void removeFeatureAt(int index);

    int getFeatureCount();

    View getFeatureAt(int index);
}

File: android/src/main/java/com/github/quadflask/react/navermap/RNNaverMapViewProps.kt
package com.github.quadflask.react.navermap

import android.view.View
import com.naver.maps.geometry.LatLng
import com.naver.maps.geometry.LatLngBounds
import com.naver.maps.map.NaverMap.MapType

interface RNNaverMapViewProps {
  fun onStop()
  fun setCenter(latLng: LatLng?)
  fun setCenter(latLng: LatLng?, zoom: Double?, tilt: Double?, bearing: Double?)
  fun zoomTo(latLngBounds: LatLngBounds, paddingInPx: Int)
  fun setTilt(tilt: Int)
  fun setBearing(bearing: Int)
  fun setZoom(zoom: Float)
  fun setMapPadding(left: Int, top: Int, right: Int, bottom: Int)
  fun onInitialized()
  fun showsMyLocationButton(show: Boolean)
  fun setCompassEnabled(show: Boolean)
  fun setScaleBarEnabled(show: Boolean)
  fun setZoomControlEnabled(show: Boolean)
  fun setLocationTrackingMode(mode: Int)
  fun setMapType(value: MapType?)
  fun setBuildingHeight(height: Float)
  fun setMinZoom(minZoomLevel: Float)
  fun setMaxZoom(maxZoomLevel: Float)
  fun setExtent(bounds: LatLngBounds?)
  fun setLayerGroupEnabled(layerGroup: String, enable: Boolean)
  fun setNightModeEnabled(enable: Boolean)
  fun setLogoMargin(left: Int, top: Int, right: Int, bottom: Int)
  fun setLogoGravity(gravity: Int)
  fun setScrollGesturesEnabled(enabled: Boolean)
  fun setZoomGesturesEnabled(enabled: Boolean)
  fun setTiltGesturesEnabled(enabled: Boolean)
  fun setRotateGesturesEnabled(enabled: Boolean)
  fun setStopGesturesEnabled(enabled: Boolean)
  fun setLiteModeEnabled(enabled: Boolean)
  fun moveCameraFitBound(bounds: LatLngBounds, left: Int, top: Int, right: Int, bottom: Int)
  fun addFeature(child: View?, index: Int)
  fun removeFeatureAt(index: Int)
  val featureCount: Int

  fun getFeatureAt(index: Int): View?

  companion object {
    val EVENT_NAMES = arrayOf(
      "onInitialized",
      "onCameraChange",
      "onMapClick",
      "onTouch",
      "onOptionChange"
    )
  }
}
